src/System.Net.Sockets/src/System/Net/Sockets/Socket.Windows.cs

```diff
@@ -166,18 +166,27 @@ internal static void SocketListToFileDescriptorSet(IList socketList, Span<IntPtr
             fileDescriptorSet[0] = (IntPtr)count;
             for (int current = 0; current < count; current++)
             {
-                if (!(socketList[current] is Socket))
+                if (!(socketList[current] is Socket socket))
                 {
                     throw new ArgumentException(SR.Format(SR.net_sockets_select, socketList[current].GetType().FullName, typeof(System.Net.Sockets.Socket).FullName), nameof(socketList));
                 }
 
-                fileDescriptorSet[current + 1] = ((Socket)socketList[current])._handle.DangerousGetHandle();
+                bool success = false;
+                socket.SafeHandle.DangerousAddRef(ref success);
+                Debug.Assert(success);
+                if (!success)
+                {
+                    throw new ObjectDisposedException(socket.SafeHandle.GetType().FullName);
+                }
```

Comment:
This if block should not be necessary.  DangerousAddRef will either set success to true or throw.

Reply:
i.e. it can just be:
```C#
bool success = false;
socket.SafeHandle.DangerousAddRef(ref success);
fileDescriptorSet[current + 1] = socket.SafeHandle.DangerousGetHandle();
```